merkledag/merkledag.go

```diff
@@ -23,8 +23,14 @@ func init() {
 	ipld.Register(cid.DagCBOR, ipldcbor.DecodeBlock)
 }
 
+// contextKey is a type to use as value for the ProgressTracker contexts.
+type contextKey string
+
+const progressContextKey contextKey = "progress"
+
 // NewDAGService constructs a new DAGService (using the default implementation).
-func NewDAGService(bs bserv.BlockService) *dagService {
```

Comment:
Does golint really complain about this?

I actually prefer to return concrete types whenever possible. It helps to deal with multiple potentially fulfilled interfaces. For example, if `*dagService` implements `X` which is a superset of `DAGService`, I can pass `NewDAGService` to something that wants an `X` without extra type assertions.

Reply:
Want me to export `dagService` instead?